google-sheets/functions/log-sms.js

```diff
@@ -0,0 +1,41 @@
+const { google } = require('googleapis');
+
+exports.handler = async function(context, event, callback) {
+  // Assemble an authentication JWT from a service account email and private
+  // key provided as environment variables:
+  const auth = new google.auth.JWT(
+    context.SHEETS_CLIENT_EMAIL,
+    null,
+    context.SHEETS_PRIVATE_KEY.replace(/\\n/g, '\n'),
+    [ 'https://www.googleapis.com/auth/spreadsheets' ],
+  );
+  const twiml = new Twilio.twiml.MessagingResponse();
+
+  try {
+    const sheets = google.sheets({
+      version: 'v4',
+      auth,
+    });
+
+    await sheets.spreadsheets.values.append({
+      spreadsheetId: context.SHEETS_DOC_ID,
+      range: `'${context.SHEETS_SHEET_NAME}'`,
+      valueInputOption: 'USER_ENTERED',
+      requestBody: {
+        values: [
+          [event.SmsSid, event.From, event.Body]
+        ],
+      }});
+
+    twiml.message('The SMS was successfully saved.');
+    callback(null, twiml);
+  } catch (error) {
+    if (error.code === 404) {
+      console.error('Could not find your Google Sheets document. Please ensure SHEETS_DOC_ID is correct.');
```

Comment:
i think these need to be twiml message responses instead of `console.error` That way if you return the twiml in the callback, whoever texted the number would receive the error message on their phone.

We should probably also have a base case error like you have in `check-sheets-config` and a test to go along with it

Reply:
I think it should be both. We should return TwiML that says that an error occurred and then guide people to the Debugger/Console logs.